esphome/components/zyaura/zyaura.cpp

```diff
@@ -0,0 +1,117 @@
+#include "zyaura.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+
+namespace esphome {
+namespace zyaura {
+
+static const char *TAG = "zyaura";
+
+ZaMessage* ZaDataProcessor::process(unsigned long ms, bool data) {
+  // check if a new message has started, based on time since previous bit
+  if ((ms - this->prev_ms_) > ZA_MAX_MS) {
+    this->num_bits_ = 0;
+  }
+  this->prev_ms_ = ms;
+
+  // number of bits received is basically the "state"
+  if (this->num_bits_ < ZA_FRAME_SIZE) {
+    // store it while it fits
+    int idx = this->num_bits_ / ZA_BITS_IN_BYTE;
```

Comment:
```suggestion
    int idx = this->num_bits_ / 8;
```

A constant for bits in a byte just over-complicates the source code.